ansible/roles/hyperthreading/tasks/main.yml

```diff
@@ -0,0 +1,54 @@
+---
+- name: disable hyper-threading
+  block:
+
```

Comment:
I didn't even realize you could name blocks until now -- makes me wonder if it actually does anything. I'll have to look.
Anyway, something that we can do to improve readability when writing conditional blocks is to put the conditions at the top so that it's obvious when reading top to bottom:
```
# Disable hyperthreading
- when: not hyperthreading_enabled
  block:
    - name: install disable-hyperthreading.sh
      copy:
        src: files/disable-hyperthreading.sh
        dest: /usr/local/bin/disable-hyperthreading.sh
        owner: root
        group: root
        mode: 0755
      notify:
        - reload and start disable-hyperthreading

[...]

# Enable hyperthreading
- when: hyperthreading_enabled
  block:
    - name: check service exists
      stat:
        path: /etc/systemd/system/disable-hyperthreading.service
      register: service_out

[...]
```